# Data Understanding

In [1]:
import pandas as pd
df = pd.read_csv('flight.csv')

df.sample(10)

MEMBER_NO    FFP_DATE FIRST_FLIGHT_DATE  GENDER  FFP_TIER  \
18894      42986   9/25/2012        10/15/2012    Male         4   
21447      40622   4/10/2010         4/10/2010    Male         4   
54249      57235    5/7/2008        12/22/2008  Female         4   
18232      57621  10/16/2008        12/17/2008    Male         4   
15225      37203   6/20/2007         7/27/2007    Male         4   
13180      57799    8/3/2009        10/18/2009    Male         4   
30670        934   9/12/2006          6/2/2011    Male         4   
52613       7454   3/15/2013         3/21/2013    Male         4   
32139      46075    1/6/2007         3/23/2007    Male         4   
33031      48274  10/10/2009        10/10/2009  Female         4   

           WORK_CITY  WORK_PROVINCE WORK_COUNTRY   AGE  LOAD_TIME  ...  \
18894        beijing        beijing           CN  39.0  3/31/2014  ...   
21447      guangzhou      guangdong           CN  36.0  3/31/2014  ...   
54249       shenzhen      guangdong           CN  34.0  3/31/2014  ...   
18232            NaN            NaN           SG  50.0  3/31/2014  ...   
15225      guangzhou      guangdong           CN  41.0  3/31/2014  ...   
13180         wuzhou        guangxi           CN  35.0  3/31/2014  ...   
30670       shanghai       shanghai           CN  53.0  3/31/2014  ...   
52613    shenyangshi  liaoningsheng           CN  57.0  3/31/2014  ...   
32139  Clayton South       Victoria           AU  33.0  3/31/2014  ...   
33031       changsha          hunan           CN  27.0  3/31/2014  ...   

       SUM_YR_2  SEG_KM_SUM  LAST_FLIGHT_DATE  LAST_TO_END  AVG_INTERVAL  \
18894    3315.0       13825         1/10/2014           82     50.222222   
21447       0.0       15751        11/25/2012          493     44.250000   
54249    1393.0        4125         6/22/2013          284    131.500000   
18232       0.0       18240        11/26/2012          492     37.000000   
15225   10709.0       22567         3/31/2014            1     22.687500   
13180   13201.0       25574         3/14/2014           18     35.388889   
30670    2125.0        9922         1/23/2014           69     64.000000   
52613    1648.0        4221         3/21/2014           11    183.000000   
32139       0.0        8801         6/24/2012          647      2.000000   
33031    1540.0       12172          7/2/2013          274     56.714286   

      MAX_INTERVAL  EXCHANGE_COUNT  avg_discount  Points_Sum  Point_NotFlight  
18894          107               0      0.936198       12323                0  
21447           93               0      0.723002        6226                0  
54249          212               0      0.514861        7516                2  
18232          123               0      0.736985        9074                0  
15225           88               2      0.697413       27115               39  
13180          168               0      0.694051       22307               16  
30670          236               0      0.730515        6378                0  
52613          273               0      0.560836        1842                0  
32139            2               0      0.768532        7538                0  
33031          256               0      0.533171        4532                0  

[10 rows x 23 columns]

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62988 entries, 0 to 62987
Data columns (total 23 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   MEMBER_NO          62988 non-null  int64  
 1   FFP_DATE           62988 non-null  object 
 2   FIRST_FLIGHT_DATE  62988 non-null  object 
 3   GENDER             62985 non-null  object 
 4   FFP_TIER           62988 non-null  int64  
 5   WORK_CITY          60719 non-null  object 
 6   WORK_PROVINCE      59740 non-null  object 
 7   WORK_COUNTRY       62962 non-null  object 
 8   AGE                62568 non-null  float64
 9   LOAD_TIME          62988 non-null  object 
 10  FLIGHT_COUNT       62988 non-null  int64  
 11  BP_SUM             62988 non-null  int64  
 12  SUM_YR_1           62437 non-null  float64
 13  SUM_YR_2           62850 non-null  float64
 14  SEG_KM_SUM         62988 non-null  int64  
 15  LAST_FLIGHT_DATE   62988 non-null  object 
 16  LAST_TO_END        629

Dataset yang Anda miliki adalah data pelanggan program Frequent Flyer, yang berisi informasi tentang perilaku dan karakteristik pelanggan. Berikut adalah penjelasan tentang dataset Anda:

Dataset ini berisi 62988 baris data dan 20 kolom, yang dapat dikategorikan menjadi beberapa bagian:

Informasi Identitas

- MEMBER_NO-b: ID unik untuk setiap pelanggan
- GENDER: Jenis kelamin pelanggan
- AGE: Umur pelanggan
- WORK_CITY, WORK_PROVINCE, WORK_COUNTRY: Informasi tentang lokasi asal pelanggan

Informasi Penerbangan

- FIRST_FLIGHT_DATE: Tanggal penerbangan pertama pelanggan
- LAST_FLIGHT_DATE: Tanggal penerbangan terakhir pelanggan
- FLIGHT_COUNT: Jumlah penerbangan yang telah dilakukan pelanggan
- SEG_KM_SUM: Total jarak (km) penerbangan yang telah dilakukan pelanggan

Informasi Program Frequent Flyer

- FFP_DATE: Tanggal bergabung dengan program Frequent Flyer
- FFP_TIER: Tier pelanggan dalam program Frequent Flyer
- BP_SUM: Rencana perjalanan pelanggan
- SUM_YR_1, SUM_YR_2: Total kredit/point di tahun pertama dan kedua
- Points_Sum: Jumlah poin yang telah didapat pelanggan
- Point_NotFlight: Poin yang tidak digunakan oleh pelanggan

Informasi Lain

- LOAD_TIME: Tanggal data diambil
- AVG_INTERVAL: Rata-rata jarak waktu antara penerbangan
- MAX_INTERVAL: Maksimal jarak waktu antara penerbangan
- EXCHANGE_COUNT: Jumlah penukaran
- avg_discount: Rata-rata diskon yang didapat pelanggan